nixjs-rt/src/builtins.ts

```diff
@@ -357,8 +358,7 @@ export function getBuiltins() {
 
       let pathValue = "";
       if (pathStrict instanceof NixString) {
-        // TODO: Get relative paths working
-        pathValue = normalizePath(pathStrict.toJs());
+        pathValue = normalizePath(joinPaths(ctx.scriptDir, pathStrict.toJs()));
```

Comment:
I wonder if all of this is actually needed. I don't think strings as relative paths are supported by Nix. Can you please double-check? It would be a shame if we had to introduce all this context-passing just for this.

Reply:
I undid the relative string thing, but are you sure there are zero builtins that rely on the current environment/ctx in any way? I'll undo if you think it's unnecessary

Reply:
Actually, thinking more about this, the eval context is already available in the lexical scope, so it's already passed everywhere (no need to add it through the lambda). If builtins ever need the context, they can grab it straight from the already available scope.

So, yes, I'm pretty sure that passing the context around like this is not necessary. The context is already there in scope.

I hope that makes sense.

Reply:
Which scope? The one created by `emit_js.rs`? That one would be a completely different file to v8, so it doesn't really work

Reply:
Let's give it a try without passing the evalCtx through here. Let's only add it when it's actually needed and there's no other way around it.